pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -354,19 +369,22 @@ func (r *reconciler) updateLoadBalancerService(current, desired *corev1.Service,
 func loadBalancerServiceChanged(current, expected *corev1.Service) (bool, *corev1.Service) {
 	updated := current.DeepCopy()
 
-	// Preserve everything but the AWS LB health check interval annotation
+	// Preserve everything but the AWS LB health check interval annotation &
+	// GCP Global Access internal Load Balancer annotation.
 	// (see <https://bugzilla.redhat.com/show_bug.cgi?id=1908758>).
 	// Updating annotations and spec fields cannot be done unless the
 	// previous release blocks upgrades when the user has modified those
 	// fields (see <https://bugzilla.redhat.com/show_bug.cgi?id=1905490>).
 	if updated.Annotations == nil {
 		updated.Annotations = map[string]string{}
 	}
-	if current.Annotations[awsLBHealthCheckIntervalAnnotation] == expected.Annotations[awsLBHealthCheckIntervalAnnotation] {
+	if current.Annotations[awsLBHealthCheckIntervalAnnotation] == expected.Annotations[awsLBHealthCheckIntervalAnnotation] &&
+		current.Annotations[GCPGlobalAccessAnnotation] == expected.Annotations[GCPGlobalAccessAnnotation] {
 		return false, nil
 	}
 
 	updated.Annotations[awsLBHealthCheckIntervalAnnotation] = expected.Annotations[awsLBHealthCheckIntervalAnnotation]
+	updated.Annotations[GCPGlobalAccessAnnotation] = expected.Annotations[GCPGlobalAccessAnnotation]
```

Comment:
Could we check the platform type and only add these annotations if the platform matches?

Reply:
We could, but we already check the platform type before setting the annotations anyways, so I think that would be redundant.